### PR TITLE
[Under FF] First version of DS Picker and DS modal

### LINKED ISSUE #65050: Refactor current DS picker to help user to find the right DS
* [Figma](https://www.figma.com/file/QGmQM1TS5ufz3u6w0H8Z2H/Panel-edit?node-id=310-9642&t=pG0isw0csvihapY4-0) 

- [ ] Implement new DS picker based on enrich dropdown

### PR DESCRIPTION
**Pull request by ivanortegaalba**
**NOTE:** This is the first version of the picker to put in ops and get early feedback. Technical refactors will be done, to improve this implementation.

https://user-images.githubusercontent.com/5699976/233039529-104e7a5b-2366-4866-868f-e126c2d52526.mp4

**Work done**
* Add a wrapper to switch between the previous and new DS picker depending on the feature toggle `advancedDataSourcePicker`.
* Add a new component to represent the modal DS picker, which we will refer as advanced DS picker
* Integrate this into the Edit panel, for now, until we're ready to replace everywhere the grafana-runtime DS picker by the wrapper.
* Replace Drawer component with the dropdown
* Adjust the first version of the styles to fit into [this Figma design](https://www.figma.com/file/QGmQM1TS5ufz3u6w0H8Z2H/Panel-edit?node-id=367-16845&t=Pk85k3mo6VyVRlAW-0)
* Adjust the design of the `FileDropzoneDefaultChildren` to match with the new DS modal but everywhere else is used nowadays.

**How to test it**
1. Important to test with the FF enabled, but also with the FF disabled
2. Make sure the new DS select properly the clicked DS
3. Make sure the upload file works properly
4. Make sure the new design of FileDrop default content fits in every place is used: import dashboard, edit panel, modal 

**Known issues to be solved**
* Accessibility of the dropdown, the user cannot access the dropdown item using the keyboard
* The height of the dropdown is not optimal, it flips too early. Probably we could adjust better, and make the dropdown grow with the screen but shrink until a certain min-height.
* The styles are not 100% accurate to the Figma file. We will work on this after the first round of feedback
* Only the edit panel is using the new component. This is on purpose. We want to test here first
* We're duplicating the usage of `getDataSourceSrv`. Here is a PR to refactor this to hook, also to simplify the component structure

### CODE REVIEW

**Review by ivanortegaalba — COMMENTED** [review]
Great additions @oscarkilhed!

I found some small things to fix
1. The input is not fully clickable

2. If the screen has a smaller height, the dropdown quickly flips vertically. I think it should reduce the height first, and then, for a minimum height, it flips. But this is something to check with Patrick.
Captura de pantalla 2023-04-17 a las 9 24 20

**Review by oscarkilhed — APPROVED** [review]
Since this is all behind a feature flag I'd say this is good to go.

File: public/app/features/datasources/components/picker/DataSourcePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import {
+  DataSourcePicker as DeprecatedDataSourcePicker,
+  DataSourcePickerProps as DeprecatedDataSourcePickerProps,
+} from '@grafana/runtime';
+import { config } from 'app/core/config';
+
+import { DataSourcePickerWithHistory } from './DataSourcePickerWithHistory';
+import { DataSourcePickerWithHistoryProps } from './types';
+
+type DataSourcePickerProps = DeprecatedDataSourcePickerProps | DataSourcePickerWithHistoryProps;
```
Comment (torkelo) [inline public/app/features/datasources/components/picker/DataSourcePicker.tsx:12]: @ivanortegaalba in what way has the DataSourcePickerProps defined in runtime changed? (this is exposed to plugins)

**Comment by aangelisc** [comment]
@ivanortegaalba it looks like this PR has broken the Azure Monitor E2E tests. It seems to break the selection of a datasource with this error: 
```
 CypressError: Timed out retrying after 30050ms: `cy.click()` failed because this element is detached from the DOM.

<div class="css-uvldi-input-suffix">...</div>
```

**Comment by gtk-grafana** [comment]
@ivanortegaalba heads up that users without a mouse are unable to select the link in the stick "footer" inside the select component.

**Pull request merged**